expression/builtin_compare.go

```diff
@@ -252,6 +248,13 @@ func (s *builtinCompareSig) eval(row []types.Datum) (d types.Datum, err error) {
 		return
 	}
 
+	if s.op != opcode.NullEQ {
+		if aa, bb, err := types.CoerceDatum(sc, a, b); err == nil {
```

Comment:
Here I think we should continue do compare with original a and b, even if coerce fail.
what do you think?